### PR TITLE
implement major changes for database importing

### PR DESCRIPTION
**Pull request by Valentin-Dumitrache**

### CODE REVIEW

File: src/pages/dish-list.vue [hunk 1]
```diff
@@ -19,20 +19,43 @@
...
 export default {
-  components: { DishCarousel },
-  computed: {
-    dishList() {
-      return dishes;
+  components: { DishCarousel, Circle2 },
+  data() {
+    return {
+      dishes: [],
+      isLoading: true
```
Comment (alina-dumitrache) [inline src/pages/dish-list.vue:37]: isLoading: false

File: src/pages/dish-list.vue [hunk 2]
```diff
@@ -19,20 +19,43 @@
...
+    };
+  },
+  async created() {
+    try {
```
Comment (alina-dumitrache) [inline src/pages/dish-list.vue:41]: Before try:
isLoading: true

File: src/pages/dish-list.vue [hunk 3]
```diff
@@ -19,20 +19,43 @@
...
+      const response = await axios.get(
+        `https://homefood-app.herokuapp.com/getDishes`
+      );
+      this.dishes = response.data;
+      this.isLoading = false;
+    } catch (e) {
+      console.log(e);
```
Comment (alina-dumitrache) [inline src/pages/dish-list.vue:48]: finally {
isLoading: false
}
(remove from try)

Comment (alina-dumitrache) [inline src/pages/dish-list.vue:48]: Remove console.log
Add TODO: error handling

File: package.json [hunk 4]
```diff
@@ -10,9 +10,11 @@
   },
   "dependencies": {
     "@fortawesome/fontawesome-free": "^5.13.0",
+    "axios": "^0.19.2",
     "core-js": "^3.6.4",
     "register-service-worker": "^1.6.2",
     "vue": "^2.6.11",
+    "vue-loading-spinner": "^1.0.11",
```
Comment (alina-dumitrache) [inline package.json:17]: Use vuetify spinner

File: src/pages/cook-profile.vue [hunk 5]
```diff
@@ -25,28 +25,43 @@
...
+  data() {
+    return {
+      dishes: [],
+      cook: [],
+      isLoading: true
+    };
+  },
+  async created() {
+    try {
+      let response = await axios.get(
```
Comment (alina-dumitrache) [inline src/pages/cook-profile.vue:52]: Move to store action and call action from store here

File: src/pages/dish-details.vue [hunk 6]
```diff
@@ -73,24 +74,57 @@
...
+      ingredients: [],
+      extraPictures: [],
+      isLoading: true
+    };
+  },
+  mixins: [icons, images],
+  async created() {
+    try {
+      let response = await axios.get(
+        `https://homefood-app.herokuapp.com/getDishForCook?id=${this.$route.params.id}`
```
Comment (alina-dumitrache) [inline src/pages/dish-details.vue:99]: Move to store action and modify endpoint

File: src/pages/dish-list.vue [hunk 7]
```diff
@@ -19,20 +19,43 @@
...
     }
   }
 };
 </script>
+<style lang="scss">
+.spinner {
```
Comment (alina-dumitrache) [inline src/pages/dish-list.vue:54]: Remove when using the Vuetify spinner

File: src/pages/cook-profile.vue [hunk 8]
```diff
@@ -1,19 +1,19 @@
 <template>
```
Comment (Valentin-Dumitrache) [inline src/pages/cook-profile.vue:1]: Please ignore this file for now as I'm waiting on feedback for the rest until I finish this.

File: src/pages/dish-list.vue (hunk 7 above)
Comment (Valentin-Dumitrache) [reply src/pages/dish-list.vue:63]: It's still needed to position.

File: src/pages/dish-details.vue [hunk 9]
```diff
@@ -1,96 +1,123 @@
...
+      cook: [],
+      ingredients: [],
+      pictures: [],
+      isLoading: false
+    };
+  },
+  mixins: [icons, images],
+  async created() {
+    this.toggleIsLoading();
+    await this.$store
```
Comment (alina-dumitrache) [inline src/pages/dish-details.vue:101]: To access store in components you need to do the following:
```js
import storeModule from './store';
import { createNamespacedHelpers } from 'vuex';
const { mapActions } = createNamespacedHelpers(storeModule.namespace); // you can also import mapState, mapGetters, mapMutations
```

inside the methods property of the Vue component import the actions you need from the store as:
```js
mapActions([ 'getDishDetails'])
```
Now you can simply call the getDishDetails inside created() as if it was a method defined in your vue component instance (using this).

!! On top of the above, on the root vue file you need to add the storeModule to the root vue component as the first property.

File: src/pages/dish-details.vue [hunk 10]
```diff
@@ -1,96 +1,123 @@
...
+      .dispatch('getDishDetails', {
+        // We will modify this when we fix routing, to properly grab the ID of the dish and find the cook id from there
+        // Until then it's functional
+        cookId: '1a',
+        dishId: '1a'
+      })
+      .then(() => {
```
Comment (alina-dumitrache) [inline src/pages/dish-details.vue:108]: use modern async await syntax in combination with try catch

File: src/store/index.js [hunk 11]
```diff
@@ -1,17 +1,69 @@
...
+          `https://homefood-app.herokuapp.com/getdishdetails?dishid=${payload.dishId}&cookid=${payload.cookId}`
+        )
+        .then(response => {
+          console.log(response);
+          commit('setDish', response.data.dish);
+          commit('setCook', response.data.cook);
+          commit('setPictures', response.data.pictures);
+          commit('setIngredients', response.data.ingredients);
+        })
+        .catch(() => {
```
Comment (alina-dumitrache) [inline src/store/index.js:64]: Same comments as for the other file in terms of store.

**Pull request merged**